R/files-validation.R

```diff
@@ -0,0 +1,154 @@
+               regexec(dictionary_version_pattern, file_names))[[1]][2]
+  
+  # Acquire version number from summary sheet
+  summary_sheet <- readxl::read_excel(file.path(
+    getwd(),
+    dictionary_path,
+    file_names
+  ),
+  sheet = odm_dictionary$summary_sheet_name)
+  # Read the first column
+  summary_versions <- summary_sheet[[1]]
+  # Strip off any NA rows
+  summary_versions <- summary_versions[!is.na(summary_versions)]
+  # Select the last version
+  summary_version <- summary_versions[[length(summary_versions)]]
+  matching_versions <- FALSE
+  # Compare if the versions match
+  if (summary_version == dictionary_file_name_version_number) {
+    matching_versions <- TRUE
+  }
+  
+  return(
+    list(
+      dictionary_name = file_names,
+      dictionary_version = summary_version,
+      matching_versions = matching_versions
+    )
+  )
+}
+
+#' Validate files sheet
+#'
+#' Validate files sheet and its internal content for proper file creation.
+#'
+#' @param dictionary_name string containing the file dictionary name.
+#' @param version string containing the dictionary version.
+#' @param dictionary_path string containing path to the dictionary directory.
+#'
+#' @return 2 lists containing csvs to export and another list containing excels to export.
+validate_files_sheet <- function(dictionary_name, version, dictionary_path) {
+  files_sheet <- readxl::read_excel(
+    file.path(
+      getwd(),
+      dictionary_path,
+      dictionary_name
+    ),
+    sheet = odm_dictionary$files_sheet_name
+  )
+  sets_sheet <- readxl::read_excel(
+    file.path(
+      getwd(),
+      dictionary_path,
+      dictionary_name
+    ),
+    sheet = odm_dictionary$sets_sheet_name
+  )
+  
+  # Remove any rows with not supported fileType.
+  files_sheet_formatted <-
+    files_sheet[files_sheet$fileType %in% files$file_type$categories, unlist(files_sheet_column_names)]
+  # insert version
+  files_sheet_formatted$name <-
+    gsub('\\{version\\}', version, files_sheet_formatted$name)
+  csv_to_extract <- c()
+  excel_to_extract <- list()
+  errors <- ""
+  for (row_index in 1:nrow(files_sheet_formatted)) {
+    working_row <- files_sheet_formatted[row_index,]
+    partID <- working_row[[files$part_ID$name]]
+    fileType <- working_row[[files$file_type$name]]
+    if (fileType == files$file_type$categories$csv) {
+      sets_info <- sets_sheet[sets_sheet$setID == partID, "partID"]
+      if (nrow(sets_info) >= 1) {
+        # Append error
+        errors <-
+          paste0(errors,
+                 " \n",
+                 partID,
+                 " is recorded for csv but is found in sets.")
+      } else{
+        csv_to_extract <- c(csv_to_extract, partID)
+      }
+    } else if (fileType == files$file_type$categories$excel) {
+      sets_info <- sets_sheet[sets_sheet$setID == partID, "partID"]
+      if (nrow(sets_info) >= 1) {
+        excel_to_extract[[partID]] <- sets_info
+      } else{
+        # Append error
+        errors <-
```

Comment:
The `partID` column for an excel sheet can also be a part, it doesn't have to be a set. https://github.com/Big-Life-Lab/PHES-ODM/blob/release-automation/specs/release-automation/release-automation-specs.qmd#L230